qt/aqt/filtered_deck.py

```diff
@@ -55,22 +55,25 @@ def __init__(
         self._desired_search_1 = search
         self._desired_search_2 = search_2
 
-        self._initial_dialog_setup()
+        self._initial_dialog_setup(deck_id)
 
         # set on successful query
         self.deck: FilteredDeckForUpdate
 
         QueryOp(
             parent=self.mw,
             op=lambda col: col.sched.get_or_create_filtered_deck(deck_id=deck_id),
-            success=self.load_deck_and_show,
+            success=lambda deck: (
+                self.load_deck_and_show(deck),  # type: ignore
+                self._after_load_deck_setup(deck_id),  # type: ignore
```

Comment:
You shouldn't need the ignore statements here - if you're getting an error about it not being able to infer the type, breaking the lambda out into a nested def should fix it.

Reply:
I also found the error weird, but I had to do it this way to avoid the update running after the setup, which would always show the option.